plugins/circleci/src/circleci-config.ts

```diff
@@ -266,29 +250,27 @@ If you would like a Tool Kit configured CircleCI config file to be generated for
 
     if (rawConfig && !isAutomatedConfig(rawConfig)) {
       const config = YAML.parse(rawConfig)
-      const { jobs, nightlyJobs } = getWorkflowJobs(config)
-      const flatJobs = jobs?.map(getJobName) ?? []
-      const flatNightlyJobs = nightlyJobs?.map(getJobName) ?? []
-      const missingJobs = state.workflows['tool-kit'].jobs
-        .map(getJobName)
-        .filter((job) => !flatJobs.includes(job))
-      const missingNightlyJobs = state.workflows.nightly.jobs
-        .map(getJobName)
-        .filter((job) => !flatNightlyJobs.includes(job))
-
-      const formatMissingHooks = (hooks: string[]): string => hooks.map(styles.hook).join(', ')
+      // Don't include any fields that are in the user's config but not in our
+      // expected config. The expected config is a subset of whatever config
+      // the user chooses, and additional fields shouldn't be flagged in the
+      // error.
+      // I'm shocked that this isn't a lodash function.
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const removeExtraFields = (expected: Record<string, any>, received: Record<string, any>) => {
+        for (const [rk, rv] of Object.entries(received)) {
+          if (!expected.hasOwnProperty(rk)) {
+            delete received[rk]
+            continue
+          }
+          if (isPlainObject(rv)) {
+            removeExtraFields(expected[rk], rv)
+          }
+        }
+      }
+      removeExtraFields(state, config)
+      const difference = diff(state, config, { expand: false })
```

Comment:
suggestion: could this YAML.stringify both objects and do a string diff? would avoid ordering issues that way

Reply:
this is a good idea! but testing it now the diffs can get really confusing when arrays are out of order. the JS object diff is a bit messy too but a bit more parseable imo and I think is a bit better at creating a minimal diff.

uuuh here
![Screenshot 2023-05-05 at 18 12 27](https://user-images.githubusercontent.com/7020796/236522993-30579656-3c14-4288-8fde-89c5975d077f.png)
vs
![Screenshot 2023-05-05 at 18 12 49](https://user-images.githubusercontent.com/7020796/236523074-99371c3f-c2e6-4213-a81d-6a20f6ff837b.png)

idk